mne/forward/_field_interpolation.py

```diff
@@ -87,6 +91,127 @@ def _compute_mapping_matrix(fmd, info):
     return mapping_mat
 
 
+def compute_virtual_evoked(evoked, from_type='mag', to_type='grad',
+                           copy=True, baseline=None, n_jobs=1):
+    """Compute virtual evoked using interpolated fields in mag/grad channels.
+
+    Parameters
+    ----------
+    evoked : instance of mne.Evoked
+        The evoked object.
+    from_type : str
+        The original channel type. It can be 'mag' or 'grad'.
+    to_type : str
+        The destination channel type. It can be 'mag' or 'grad'.
+    copy : bool
+        If False evoked is modified in place.
+    n_jobs : int
+        Number of jobs to run in parallel.
+
+    Returns
+    -------
+    evoked : instance of mne.Evoked
+        The transformed evoked object containing only virtual channels.
+    """
+    if from_type not in ['mag', 'grad']:
+        raise ValueError('from_type must be "mag" or "grad", not "%s"'
+                         % from_type)
+    if to_type not in ['mag', 'grad']:
+        raise ValueError('to_type must be "mag" or "grad", not "%s"'
+                         % to_type)
+    if from_type == to_type:
+        raise ValueError('from_type and to_type cannot be the same.')
+
+    # get surface in head coordinates
+    surf_virt = get_meg_helmet_surf(evoked.info)
+
+    # pick the original and destination channels
+    pick_from = pick_types(evoked.info, meg=from_type, eeg=False,
+                           ref_meg=False)
+    pick_to = pick_types(evoked.info, meg=to_type, eeg=False,
```

Comment:
On Sat, Apr 18, 2015 at 1:54 AM, Eric Larson <notifications@github.com>
wrote:

> In mne/forward/_field_interpolation.py
> <https://github.com/mne-tools/mne-python/pull/1943#discussion_r28638165>:
>
> > +    if from_type not in ['mag', 'grad']:
> > +        raise ValueError('from_type must be "mag" or "grad", not "%s"'
> > +                         % from_type)
> > +    if to_type not in ['mag', 'grad']:
> > +        raise ValueError('to_type must be "mag" or "grad", not "%s"'
> > +                         % to_type)
> > +    if from_type == to_type:
> > +        raise ValueError('from_type and to_type cannot be the same.')
> > +
> > +    # get surface in head coordinates
> > +    surf_virt = get_meg_helmet_surf(evoked.info)
> > +
> > +    # pick the original and destination channels
> > +    pick_from = pick_types(evoked.info, meg=from_type, eeg=False,
> > +                           ref_meg=False)
> > +    pick_to = pick_types(evoked.info, meg=to_type, eeg=False,
>
> (we an assume Matti got the physics right for now :) )
>
of course :)

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/1943/files#r28638165>.
>